python/src/scipp/constants/__init__.py

```diff
@@ -0,0 +1,106 @@
+``alpha``                    fine-structure constant
+``fine_structure``           fine-structure constant
+``N_A``                      Avogadro constant
+``Avogadro``                 Avogadro constant
+``k``                        Boltzmann constant
+``Boltzmann``                Boltzmann constant
+``sigma``                    Stefan-Boltzmann constant :math:`\sigma`
+``Stefan_Boltzmann``         Stefan-Boltzmann constant :math:`\sigma`
+``Wien``                     Wien displacement law constant
+``Rydberg``                  Rydberg constant
+``m_e``                      electron mass
+``electron_mass``            electron mass
+``m_p``                      proton mass
+``proton_mass``              proton mass
+``m_n``                      neutron mass
+``neutron_mass``             neutron mass
+===========================  =============================================
+
+In addition to the above variables, :mod:`scipp.constants` also contains the
+2018 CODATA recommended values [CODATA2018]_ database containing more physical
+constants.
+The database is accessed using :py:func:`scipp.constants.physical_constants`.
+
+.. [CODATA2018] CODATA Recommended Values of the Fundamental
+   Physical Constants 2018.
+   https://physics.nist.gov/cuu/Constants/
+"""
+import math as _math
+from .. import scalar, Variable
+
+
+def physical_constants(key: str, with_variance: bool = False) -> Variable:
+    """
+    Returns the CODATA recommended value with unit of the requested physical constant.
+
+    :param key: Key of the requested constant. See `scipy.constants.physical_constants <https://docs.scipy.org/doc/scipy/reference/constants.html#scipy.constants.physical_constants>`_ for an overview.
+    :param with_variance: Optional, if True, the uncertainty if the constant is
+                          included as the variance. Default is False.
+    :returns: Scalar variable with unit and optional variance.
+    :rtype: Variable
+    """
+    from scipy.constants import physical_constants as _cd
+    value, unit, uncertainty = _cd[key]
+    args = {'value': value, 'unit': unit.replace(' ', '*')}
+    if with_variance:
+        stddev = uncertainty
+        args['variance'] = stddev * stddev
+    return scalar(**args)
+
+
+# mathematical constants
+pi = scalar(_math.pi)
+golden = golden_ratio = scalar((1 + _math.sqrt(5)) / 2)
+
+# physical constants
+c = speed_of_light = physical_constants('speed of light in vacuum')
+mu_0 = physical_constants('vacuum mag. permeability')
+epsilon_0 = physical_constants('vacuum electric permittivity')
+h = Planck = physical_constants('Planck constant')
+hbar = h / (2 * pi)
+G = gravitational_constant = physical_constants('Newtonian constant of gravitation')
+g = physical_constants('standard acceleration of gravity')
+e = elementary_charge = physical_constants('elementary charge')
+R = gas_constant = physical_constants('molar gas constant')
+alpha = fine_structure = physical_constants('fine-structure constant')
```

Comment:
I would say `alpha` might be a bit dangerous, as it is used to represent so many different quantities in all branches of physics. I think `fine_structure` is very non-ambiguous, and we could just keep that one?

Reply:
I don't see a reason for changing anything here compared to `scipy`, unless there is a really good motivation.